### PR TITLE
Assignment4 alan and neriah

### PR DESCRIPTION
**Pull request by arose-13**
Alan (@arose-13) and Neriah (@schemcharger)

Front end isn't quite done yet. Back end is set up, but not linked to the front end yet.

### CODE REVIEW

**Review by VForev — COMMENTED** [review]
Make all pages responsive ✅
1080p desktop ✅
iPad Air ✅
Samsung Galaxy S20 Ultra Super Mega Excellent ✅
IPhone SE ✅
Create a improved landing page and site theme (Background, make it look attractive to play) ✅
Theme should extend all pages ✅
Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌

**(Word of the day still just generates a random word)**

Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌
It should indicate if the user has played that day ❌
Clicking on the item should take the player to that game (Extra Credit) ❌

Api works like it's supposed to, with no problems, just need to finish up some things.

**Comment by AndrewKorchemniy** [comment]
Will review.

File: Wordle.Api/Wordle.Api/Data/PlayedGame.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Wordle.Api.Data;
+
+public class PlayedGame
+{
+    public int Id { get; set; }
+    public DateTime Date { get; set; }
+    public string Word { get; set; }
```
Comment (AndrewKorchemniy) [inline Wordle.Api/Wordle.Api/Data/PlayedGame.cs:7]: Does PlayedGame need to know the word - or just the date?

File: Wordle.Api/Wordle.Api/Services/WordService.cs [hunk 2]
```diff
@@ -1,65 +1,144 @@
...
+    {
+        _db = db;
+        PreviousWords = new List<string>();
+        TodaysWord = new Word { Text = GetRandomWord().Result };
+        StartTimer();
+    }
+
+    public async Task<string> GetRandomWord()
+    {
+        if (TodaysWord.Text.IsNullOrEmpty())
```
Comment (AndrewKorchemniy) [inline Wordle.Api/Wordle.Api/Services/WordService.cs:25]: Getting an exception here when running locally. Several endpoints seem to cause the exception. (all from /word/*)

File: wordle-web/src/App.vue [hunk 3]
```diff
@@ -22,14 +22,29 @@
...
+              </v-list-item-title>
+            </v-list-item>
+            <v-list-item>
+              <v-list-item-title>
+                <RouterLink to="/dailywordle">Todays Word</RouterLink>
+              </v-list-item-title>
+            </v-list-item>
+            <v-list-item>
+              <v-list-item-title>
+                <RouterLink to="/previouswordles">Daily Words</RouterLink>
```
Comment (AndrewKorchemniy) [inline wordle-web/src/App.vue:35]: Buttons to leaderboard and daily word could also go on the landing page.

**Review by AndrewKorchemniy — COMMENTED** [review]
## Features

- Make all pages responsive ✅
  - 1080p desktop ✅
  - iPad Air ✅
  - Samsung Galaxy S20 Ultra Super Mega Excellent ✅
  - IPhone SE ✅ (there's some text that's being truncated though)
- Create a improved landing page and site theme (Background, make it look attractive to play) ❌
  - Theme should extend all pages ❌
- Add an instructions page that is available via the hamburger menu that details the features of the game and how to play ✅
- Create a new Wordle game page that reuses components but will provide the same word for a specific day. Everyone going to the page that day will get the same word. The page should indicate Wordle of the Day or something like that ❌
- Create a page that lists the last 10 daily words with statistics. Number of plays, Average score, Average time. (just list the date, don't show the word silly) ❌✅(wip)
  - It should indicate if the user has played that day ❌
  - Clicking on the item should take the player to that game (Extra Credit) ❌

## Comments

- Instructions page: I think the instructions themselves are good enough, the styling doesn't match any theme or background though. You guys could also place a button within the v-card actions to go back to the game. p.s. a dialog utilizing a card might be easier and more appealing.
- Backend: It's mostly complete. I was getting an exception on GetRandomWord (/Word, (WordService.cs.25)), so I could only test it from swagger.
- Landing page: I recommend taking a look at [nytime's wordle](https://www.nytimes.com/games/wordle/index.html) for inspiration. Vera and I did something similar.

File: wordle-web/src/App.vue [hunk 4]
```diff
@@ -22,14 +22,29 @@
           <v-list>
             <v-list-item>
               <v-list-item-title>
-                <RouterLink to="/">Play Again</RouterLink>
+                <RouterLink to="/game">Play Game</RouterLink>
+              </v-list-item-title>
+            </v-list-item>
+            <v-list-item>
+              <v-list-item-title>
+                <RouterLink to="/dailywordle">Todays Word</RouterLink>
```
Comment (mmwoodfo) [inline wordle-web/src/App.vue:30]:
```suggestion
                <RouterLink to="/dailywordle">Daily Word</RouterLink>
```
I'd try and match the route name to the navigation item name

File: wordle-web/src/views/DailyView.vue [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+// Add this to make testing work because useDisplay() throws an error when testing
+// Wrap useDisplay in a function so that it doesn't get called during testing.
+const display = inject(Services.Display, () => reactive(useDisplay())) as unknown as ReturnType<
+  typeof useDisplay
+>
+const playerService = inject(Services.PlayerService) as PlayerService
+
+onMounted(async () => {
+  // Start a new game
+  await newGame()
```
Comment (mmwoodfo) [inline wordle-web/src/views/DailyView.vue:93]: This ends up awaiting for ever

File: wordle-web/src/views/PreviousView.vue [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      <v-col>{{ day.averageAttempts }}</v-col>
+    </v-row>
+    <v-card-text> </v-card-text>
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import { previousWords } from '@/scripts/previousWords'
+
+const previous = previousWords
```
Comment (mmwoodfo) [inline wordle-web/src/views/PreviousView.vue:15]: I don't see any data being populated on this page

**Review by mmwoodfo — COMMENTED** [review]
Do you have a link to your website?

**Pull request closed without merging**